From 1f8fa865312badab956c71c05dcd54e041bb446d Mon Sep 17 00:00:00 2001
From: "Alejandro G. Castro" <alex@igalia.com>
Date: Thu, 3 Sep 2026 22:54:53 +0200
Subject: [PATCH] SVG property registry should not take the initial value as a non-type template parameter

https://bugs.webkit.org/show_bug.cgi?id=323357
Upstream PR: https://github.com/WebKit/WebKit/pull/73185

320226@main started storing each SVGAnimatedInteger and SVGAnimatedNumber
attribute's initial value on its accessor, and passed it as a non-type template
parameter alongside the property. On the other hand initialValue does not need
to be a template parameter. The singleton is already unique per property, and a
property is registered exactly once.

This is patch is asimplification of the original without changing the idea behind
the original patch. We detected this issue compiling for and older Android NDK.

No behaviour change so original tests should check if the patch is
correct.

* Source/WebCore/svg/SVGComponentTransferFunctionElement.cpp:
(WebCore::SVGComponentTransferFunctionElement::PropertyRegistry::registerProperties):
* Source/WebCore/svg/SVGFEConvolveMatrixElement.cpp:
(WebCore::SVGFEConvolveMatrixElement::PropertyRegistry::registerProperties):
* Source/WebCore/svg/SVGFEDiffuseLightingElement.cpp:
(WebCore::SVGFEDiffuseLightingElement::PropertyRegistry::registerProperties):
* Source/WebCore/svg/SVGFEDropShadowElement.cpp:
(WebCore::SVGFEDropShadowElement::PropertyRegistry::registerProperties):
* Source/WebCore/svg/SVGFELightElement.cpp:
(WebCore::SVGFELightElement::PropertyRegistry::registerProperties):
* Source/WebCore/svg/SVGFESpecularLightingElement.cpp:
(WebCore::SVGFESpecularLightingElement::PropertyRegistry::registerProperties):
* Source/WebCore/svg/SVGFETurbulenceElement.cpp:
(WebCore::SVGFETurbulenceElement::PropertyRegistry::registerProperties):
* Source/WebCore/svg/properties/SVGAnimatedPrimitivePropertyAccessor.h:
* Source/WebCore/svg/properties/SVGAnimatedPropertyAccessorImpl.h:
* Source/WebCore/svg/properties/SVGAnimatedPropertyPairAccessor.h:
* Source/WebCore/svg/properties/SVGAnimatedPropertyPairAccessorImpl.h:
* Source/WebCore/svg/properties/SVGPropertyOwnerRegistry.h:
---
 .../SVGComponentTransferFunctionElement.cpp   |  6 ++---
 .../svg/SVGFEConvolveMatrixElement.cpp        |  4 ++--
 .../svg/SVGFEDiffuseLightingElement.cpp       |  4 ++--
 Source/WebCore/svg/SVGFEDropShadowElement.cpp |  6 ++---
 Source/WebCore/svg/SVGFELightElement.cpp      |  2 +-
 .../svg/SVGFESpecularLightingElement.cpp      |  6 ++---
 Source/WebCore/svg/SVGFETurbulenceElement.cpp |  2 +-
 .../SVGAnimatedPrimitivePropertyAccessor.h    | 12 ++++++----
 .../SVGAnimatedPropertyAccessorImpl.h         |  8 +++----
 .../SVGAnimatedPropertyPairAccessor.h         |  7 +++---
 .../SVGAnimatedPropertyPairAccessorImpl.h     |  8 +++----
 .../svg/properties/SVGPropertyOwnerRegistry.h | 24 +++++++++----------
 12 files changed, 46 insertions(+), 43 deletions(-)

diff --git a/Source/WebCore/svg/SVGComponentTransferFunctionElement.cpp b/Source/WebCore/svg/SVGComponentTransferFunctionElement.cpp
index f21235d59191..e6d55b2bdb42 100644
--- a/Source/WebCore/svg/SVGComponentTransferFunctionElement.cpp
+++ b/Source/WebCore/svg/SVGComponentTransferFunctionElement.cpp
@@ -43,10 +43,10 @@ SVGComponentTransferFunctionElement::SVGComponentTransferFunctionElement(const Q
         didRegistration = true;
         PropertyRegistry::registerProperty<SVGNames::typeAttr, ComponentTransferType, &SVGComponentTransferFunctionElement::m_type>();
         PropertyRegistry::registerProperty<SVGNames::tableValuesAttr, &SVGComponentTransferFunctionElement::m_tableValues>();
-        PropertyRegistry::registerProperty<SVGNames::slopeAttr, &SVGComponentTransferFunctionElement::m_slope, initialSlopeValue>();
+        PropertyRegistry::registerProperty<SVGNames::slopeAttr, &SVGComponentTransferFunctionElement::m_slope>(initialSlopeValue);
         PropertyRegistry::registerProperty<SVGNames::interceptAttr, &SVGComponentTransferFunctionElement::m_intercept>();
-        PropertyRegistry::registerProperty<SVGNames::amplitudeAttr, &SVGComponentTransferFunctionElement::m_amplitude, initialAmplitudeValue>();
-        PropertyRegistry::registerProperty<SVGNames::exponentAttr, &SVGComponentTransferFunctionElement::m_exponent, initialExponentValue>();
+        PropertyRegistry::registerProperty<SVGNames::amplitudeAttr, &SVGComponentTransferFunctionElement::m_amplitude>(initialAmplitudeValue);
+        PropertyRegistry::registerProperty<SVGNames::exponentAttr, &SVGComponentTransferFunctionElement::m_exponent>(initialExponentValue);
         PropertyRegistry::registerProperty<SVGNames::offsetAttr, &SVGComponentTransferFunctionElement::m_offset>();
     };
 }
diff --git a/Source/WebCore/svg/SVGFEConvolveMatrixElement.cpp b/Source/WebCore/svg/SVGFEConvolveMatrixElement.cpp
index e023283ddd42..f90161be8c17 100644
--- a/Source/WebCore/svg/SVGFEConvolveMatrixElement.cpp
+++ b/Source/WebCore/svg/SVGFEConvolveMatrixElement.cpp
@@ -46,9 +46,9 @@ inline SVGFEConvolveMatrixElement::SVGFEConvolveMatrixElement(const QualifiedNam
     if (!didRegistration) [[unlikely]] {
         didRegistration = true;
         PropertyRegistry::registerProperty<SVGNames::inAttr, &SVGFEConvolveMatrixElement::m_in1>();
-        PropertyRegistry::registerProperty<SVGNames::orderAttr, &SVGFEConvolveMatrixElement::m_orderX, &SVGFEConvolveMatrixElement::m_orderY, initialOrderValue, initialOrderValue>();
+        PropertyRegistry::registerProperty<SVGNames::orderAttr, &SVGFEConvolveMatrixElement::m_orderX, &SVGFEConvolveMatrixElement::m_orderY>(initialOrderValue, initialOrderValue);
         PropertyRegistry::registerProperty<SVGNames::kernelMatrixAttr, &SVGFEConvolveMatrixElement::m_kernelMatrix>();
-        PropertyRegistry::registerProperty<SVGNames::divisorAttr, &SVGFEConvolveMatrixElement::m_divisor, initialDivisorValue>();
+        PropertyRegistry::registerProperty<SVGNames::divisorAttr, &SVGFEConvolveMatrixElement::m_divisor>(initialDivisorValue);
         PropertyRegistry::registerProperty<SVGNames::biasAttr, &SVGFEConvolveMatrixElement::m_bias>();
         PropertyRegistry::registerProperty<SVGNames::targetXAttr, &SVGFEConvolveMatrixElement::m_targetX>();
         PropertyRegistry::registerProperty<SVGNames::targetYAttr, &SVGFEConvolveMatrixElement::m_targetY>();
diff --git a/Source/WebCore/svg/SVGFEDiffuseLightingElement.cpp b/Source/WebCore/svg/SVGFEDiffuseLightingElement.cpp
index 1ec5396af7b7..ff4018704835 100644
--- a/Source/WebCore/svg/SVGFEDiffuseLightingElement.cpp
+++ b/Source/WebCore/svg/SVGFEDiffuseLightingElement.cpp
@@ -45,8 +45,8 @@ inline SVGFEDiffuseLightingElement::SVGFEDiffuseLightingElement(const QualifiedN
     if (!didRegistration) [[unlikely]] {
         didRegistration = true;
         PropertyRegistry::registerProperty<SVGNames::inAttr, &SVGFEDiffuseLightingElement::m_in1>();
-        PropertyRegistry::registerProperty<SVGNames::diffuseConstantAttr, &SVGFEDiffuseLightingElement::m_diffuseConstant, initialDiffuseConstantValue>();
-        PropertyRegistry::registerProperty<SVGNames::surfaceScaleAttr, &SVGFEDiffuseLightingElement::m_surfaceScale, initialSurfaceScaleValue>();
+        PropertyRegistry::registerProperty<SVGNames::diffuseConstantAttr, &SVGFEDiffuseLightingElement::m_diffuseConstant>(initialDiffuseConstantValue);
+        PropertyRegistry::registerProperty<SVGNames::surfaceScaleAttr, &SVGFEDiffuseLightingElement::m_surfaceScale>(initialSurfaceScaleValue);
         PropertyRegistry::registerProperty<SVGNames::kernelUnitLengthAttr, &SVGFEDiffuseLightingElement::m_kernelUnitLengthX, &SVGFEDiffuseLightingElement::m_kernelUnitLengthY>();
     }
 }
diff --git a/Source/WebCore/svg/SVGFEDropShadowElement.cpp b/Source/WebCore/svg/SVGFEDropShadowElement.cpp
index d0a3f16e2d13..cd792b9468b8 100644
--- a/Source/WebCore/svg/SVGFEDropShadowElement.cpp
+++ b/Source/WebCore/svg/SVGFEDropShadowElement.cpp
@@ -45,9 +45,9 @@ inline SVGFEDropShadowElement::SVGFEDropShadowElement(const QualifiedName& tagNa
     if (!didRegistration) [[unlikely]] {
         didRegistration = true;
         PropertyRegistry::registerProperty<SVGNames::inAttr, &SVGFEDropShadowElement::m_in1>();
-        PropertyRegistry::registerProperty<SVGNames::dxAttr, &SVGFEDropShadowElement::m_dx, initialDxValue>();
-        PropertyRegistry::registerProperty<SVGNames::dyAttr, &SVGFEDropShadowElement::m_dy, initialDyValue>();
-        PropertyRegistry::registerProperty<SVGNames::stdDeviationAttr, &SVGFEDropShadowElement::m_stdDeviationX, &SVGFEDropShadowElement::m_stdDeviationY, initialStdDeviationValue, initialStdDeviationValue>();
+        PropertyRegistry::registerProperty<SVGNames::dxAttr, &SVGFEDropShadowElement::m_dx>(initialDxValue);
+        PropertyRegistry::registerProperty<SVGNames::dyAttr, &SVGFEDropShadowElement::m_dy>(initialDyValue);
+        PropertyRegistry::registerProperty<SVGNames::stdDeviationAttr, &SVGFEDropShadowElement::m_stdDeviationX, &SVGFEDropShadowElement::m_stdDeviationY>(initialStdDeviationValue, initialStdDeviationValue);
     }
 }
 
diff --git a/Source/WebCore/svg/SVGFELightElement.cpp b/Source/WebCore/svg/SVGFELightElement.cpp
index c39846483022..c583482c79a9 100644
--- a/Source/WebCore/svg/SVGFELightElement.cpp
+++ b/Source/WebCore/svg/SVGFELightElement.cpp
@@ -58,7 +58,7 @@ SVGFELightElement::SVGFELightElement(const QualifiedName& tagName, Document& doc
         PropertyRegistry::registerProperty<SVGNames::pointsAtXAttr, &SVGFELightElement::m_pointsAtX>();
         PropertyRegistry::registerProperty<SVGNames::pointsAtYAttr, &SVGFELightElement::m_pointsAtY>();
         PropertyRegistry::registerProperty<SVGNames::pointsAtZAttr, &SVGFELightElement::m_pointsAtZ>();
-        PropertyRegistry::registerProperty<SVGNames::specularExponentAttr, &SVGFELightElement::m_specularExponent, initialSpecularExponentValue>();
+        PropertyRegistry::registerProperty<SVGNames::specularExponentAttr, &SVGFELightElement::m_specularExponent>(initialSpecularExponentValue);
         PropertyRegistry::registerProperty<SVGNames::limitingConeAngleAttr, &SVGFELightElement::m_limitingConeAngle>();
     }
 }
diff --git a/Source/WebCore/svg/SVGFESpecularLightingElement.cpp b/Source/WebCore/svg/SVGFESpecularLightingElement.cpp
index eeecb6099a6b..c6f3608bd331 100644
--- a/Source/WebCore/svg/SVGFESpecularLightingElement.cpp
+++ b/Source/WebCore/svg/SVGFESpecularLightingElement.cpp
@@ -47,9 +47,9 @@ inline SVGFESpecularLightingElement::SVGFESpecularLightingElement(const Qualifie
     if (!didRegistration) [[unlikely]] {
         didRegistration = true;
         PropertyRegistry::registerProperty<SVGNames::inAttr, &SVGFESpecularLightingElement::m_in1>();
-        PropertyRegistry::registerProperty<SVGNames::specularConstantAttr, &SVGFESpecularLightingElement::m_specularConstant, initialSpecularConstantValue>();
-        PropertyRegistry::registerProperty<SVGNames::specularExponentAttr, &SVGFESpecularLightingElement::m_specularExponent, initialSpecularExponentValue>();
-        PropertyRegistry::registerProperty<SVGNames::surfaceScaleAttr, &SVGFESpecularLightingElement::m_surfaceScale, initialSurfaceScaleValue>();
+        PropertyRegistry::registerProperty<SVGNames::specularConstantAttr, &SVGFESpecularLightingElement::m_specularConstant>(initialSpecularConstantValue);
+        PropertyRegistry::registerProperty<SVGNames::specularExponentAttr, &SVGFESpecularLightingElement::m_specularExponent>(initialSpecularExponentValue);
+        PropertyRegistry::registerProperty<SVGNames::surfaceScaleAttr, &SVGFESpecularLightingElement::m_surfaceScale>(initialSurfaceScaleValue);
         PropertyRegistry::registerProperty<SVGNames::kernelUnitLengthAttr, &SVGFESpecularLightingElement::m_kernelUnitLengthX, &SVGFESpecularLightingElement::m_kernelUnitLengthY>();
     }
 }
diff --git a/Source/WebCore/svg/SVGFETurbulenceElement.cpp b/Source/WebCore/svg/SVGFETurbulenceElement.cpp
index dc532c1b6479..ea2f9a15ce65 100644
--- a/Source/WebCore/svg/SVGFETurbulenceElement.cpp
+++ b/Source/WebCore/svg/SVGFETurbulenceElement.cpp
@@ -44,7 +44,7 @@ inline SVGFETurbulenceElement::SVGFETurbulenceElement(const QualifiedName& tagNa
     if (!didRegistration) [[unlikely]] {
         didRegistration = true;
         PropertyRegistry::registerProperty<SVGNames::baseFrequencyAttr, &SVGFETurbulenceElement::m_baseFrequencyX, &SVGFETurbulenceElement::m_baseFrequencyY>();
-        PropertyRegistry::registerProperty<SVGNames::numOctavesAttr, &SVGFETurbulenceElement::m_numOctaves, initialOctavesValue>();
+        PropertyRegistry::registerProperty<SVGNames::numOctavesAttr, &SVGFETurbulenceElement::m_numOctaves>(initialOctavesValue);
         PropertyRegistry::registerProperty<SVGNames::seedAttr, &SVGFETurbulenceElement::m_seed>();
         PropertyRegistry::registerProperty<SVGNames::stitchTilesAttr, SVGStitchOptions, &SVGFETurbulenceElement::m_stitchTiles>();
         PropertyRegistry::registerProperty<SVGNames::typeAttr, TurbulenceType, &SVGFETurbulenceElement::m_type>();
diff --git a/Source/WebCore/svg/properties/SVGAnimatedPrimitivePropertyAccessor.h b/Source/WebCore/svg/properties/SVGAnimatedPrimitivePropertyAccessor.h
index 54c8ce33676b..1466007d7228 100644
--- a/Source/WebCore/svg/properties/SVGAnimatedPrimitivePropertyAccessor.h
+++ b/Source/WebCore/svg/properties/SVGAnimatedPrimitivePropertyAccessor.h
@@ -49,11 +49,13 @@ public:
     }
 
 protected:
-    // property and initialValue have to stay template arguments: they are what gives each
-    // attribute its own singleton. As function arguments there would be one per OwnerType, and the
-    // second attribute registered would silently be handed the first one's accessor.
-    template<typename AccessorType, const Ref<AnimatedPropertyType> OwnerType::*property, ValueType initialValue>
-    static const SVGMemberAccessor<OwnerType>& singleton()
+    // property has to stay a template argument: it is what gives each attribute its own
+    // singleton. As a function argument there would be one per OwnerType, and the second
+    // attribute registered would silently be handed the first one's accessor. initialValue does
+    // not have to be one: the singleton is already unique per property, and each property is
+    // registered exactly once, so the first call is the only one.
+    template<typename AccessorType, const Ref<AnimatedPropertyType> OwnerType::*property>
+    static const SVGMemberAccessor<OwnerType>& singleton(ValueType initialValue)
     {
         static NeverDestroyed<AccessorType> propertyAccessor { property, initialValue };
         return propertyAccessor;
diff --git a/Source/WebCore/svg/properties/SVGAnimatedPropertyAccessorImpl.h b/Source/WebCore/svg/properties/SVGAnimatedPropertyAccessorImpl.h
index 5ff3c3695f6d..ab4bbda16d8d 100644
--- a/Source/WebCore/svg/properties/SVGAnimatedPropertyAccessorImpl.h
+++ b/Source/WebCore/svg/properties/SVGAnimatedPropertyAccessorImpl.h
@@ -94,8 +94,8 @@ class SVGAnimatedIntegerAccessor final : public SVGAnimatedPrimitivePropertyAcce
 public:
     using Base::Base;
     using Base::property;
-    template<auto property, int initialValue>
-    constexpr static const SVGMemberAccessor<OwnerType>& singleton() { return Base::template singleton<SVGAnimatedIntegerAccessor, property, initialValue>(); }
+    template<auto property>
+    static const SVGMemberAccessor<OwnerType>& singleton(int initialValue) { return Base::template singleton<SVGAnimatedIntegerAccessor, property>(initialValue); }
 
 private:
     RefPtr<SVGAttributeAnimator> createAnimator(OwnerType& owner, const QualifiedName& attributeName, AnimationMode animationMode, CalcMode calcMode, bool isAccumulated, bool isAdditive) const final
@@ -163,8 +163,8 @@ class SVGAnimatedNumberAccessor final : public SVGAnimatedPrimitivePropertyAcces
 public:
     using Base::Base;
     using Base::property;
-    template<auto property, float initialValue>
-    constexpr static const SVGMemberAccessor<OwnerType>& singleton() { return Base::template singleton<SVGAnimatedNumberAccessor, property, initialValue>(); }
+    template<auto property>
+    static const SVGMemberAccessor<OwnerType>& singleton(float initialValue) { return Base::template singleton<SVGAnimatedNumberAccessor, property>(initialValue); }
 
 private:
     RefPtr<SVGAttributeAnimator> createAnimator(OwnerType& owner, const QualifiedName& attributeName, AnimationMode animationMode, CalcMode calcMode, bool isAccumulated, bool isAdditive) const final
diff --git a/Source/WebCore/svg/properties/SVGAnimatedPropertyPairAccessor.h b/Source/WebCore/svg/properties/SVGAnimatedPropertyPairAccessor.h
index a3b6d376d14b..d56c3e70612c 100644
--- a/Source/WebCore/svg/properties/SVGAnimatedPropertyPairAccessor.h
+++ b/Source/WebCore/svg/properties/SVGAnimatedPropertyPairAccessor.h
@@ -57,9 +57,10 @@ protected:
         return propertyAccessor;
     }
 
-    // Template arguments for the reason given in SVGAnimatedPrimitivePropertyAccessor::singleton().
-    template<typename AccessorType, auto property1, auto property2, auto initialValue1, auto initialValue2>
-    static SVGMemberAccessor<OwnerType>& singleton()
+    // The properties are template arguments for the reason given in
+    // SVGAnimatedPrimitivePropertyAccessor::singleton(); the initial values need not be.
+    template<typename AccessorType, auto property1, auto property2, typename ValueType1, typename ValueType2>
+    static SVGMemberAccessor<OwnerType>& singleton(ValueType1 initialValue1, ValueType2 initialValue2)
     {
         static NeverDestroyed<AccessorType> propertyAccessor { property1, initialValue1, property2, initialValue2 };
         return propertyAccessor;
diff --git a/Source/WebCore/svg/properties/SVGAnimatedPropertyPairAccessorImpl.h b/Source/WebCore/svg/properties/SVGAnimatedPropertyPairAccessorImpl.h
index dcc364509e96..f29e461e4da3 100644
--- a/Source/WebCore/svg/properties/SVGAnimatedPropertyPairAccessorImpl.h
+++ b/Source/WebCore/svg/properties/SVGAnimatedPropertyPairAccessorImpl.h
@@ -92,8 +92,8 @@ class SVGAnimatedIntegerPairAccessor final : public SVGAnimatedPropertyPairAcces
 
 public:
     using Base::Base;
-    template<auto property1, auto property2, int initialValue1, int initialValue2>
-    constexpr static const SVGMemberAccessor<OwnerType>& singleton() { return Base::template singleton<SVGAnimatedIntegerPairAccessor, property1, property2, initialValue1, initialValue2>(); }
+    template<auto property1, auto property2>
+    static const SVGMemberAccessor<OwnerType>& singleton(int initialValue1, int initialValue2) { return Base::template singleton<SVGAnimatedIntegerPairAccessor, property1, property2>(initialValue1, initialValue2); }
 
 private:
     std::optional<String> synchronize(const OwnerType& owner) const final
@@ -127,8 +127,8 @@ class SVGAnimatedNumberPairAccessor final : public SVGAnimatedPropertyPairAccess
 
 public:
     using Base::Base;
-    template<auto property1, auto property2, float initialValue1, float initialValue2>
-    constexpr static const SVGMemberAccessor<OwnerType>& singleton() { return Base::template singleton<SVGAnimatedNumberPairAccessor, property1, property2, initialValue1, initialValue2>(); }
+    template<auto property1, auto property2>
+    static const SVGMemberAccessor<OwnerType>& singleton(float initialValue1, float initialValue2) { return Base::template singleton<SVGAnimatedNumberPairAccessor, property1, property2>(initialValue1, initialValue2); }
 
 private:
     std::optional<String> synchronize(const OwnerType& owner) const final
diff --git a/Source/WebCore/svg/properties/SVGPropertyOwnerRegistry.h b/Source/WebCore/svg/properties/SVGPropertyOwnerRegistry.h
index a29451de2e05..be19f6c16200 100644
--- a/Source/WebCore/svg/properties/SVGPropertyOwnerRegistry.h
+++ b/Source/WebCore/svg/properties/SVGPropertyOwnerRegistry.h
@@ -95,10 +95,10 @@ public:
 
     // initialValue has to agree with what the element passes to SVGAnimatedInteger::create(): that
     // is what the property is born with, and this is only the way back to it.
-    template<const LazyNeverDestroyed<const QualifiedName>& attributeName, const Ref<SVGAnimatedInteger> OwnerType::*property, int initialValue = 0>
-    static void registerProperty()
+    template<const LazyNeverDestroyed<const QualifiedName>& attributeName, const Ref<SVGAnimatedInteger> OwnerType::*property>
+    static void registerProperty(int initialValue = 0)
     {
-        registerProperty(attributeName, SVGAnimatedIntegerAccessor<OwnerType>::template singleton<property, initialValue>());
+        registerProperty(attributeName, SVGAnimatedIntegerAccessor<OwnerType>::template singleton<property>(initialValue));
     }
 
     template<const LazyNeverDestroyed<const QualifiedName>& attributeName, const Ref<SVGAnimatedLength> OwnerType::*property>
@@ -113,10 +113,10 @@ public:
         registerProperty(attributeName, SVGAnimatedLengthListAccessor<OwnerType>::template singleton<property>());
     }
 
-    template<const LazyNeverDestroyed<const QualifiedName>& attributeName, const Ref<SVGAnimatedNumber> OwnerType::*property, float initialValue = 0.0f>
-    static void registerProperty()
+    template<const LazyNeverDestroyed<const QualifiedName>& attributeName, const Ref<SVGAnimatedNumber> OwnerType::*property>
+    static void registerProperty(float initialValue = 0)
     {
-        registerProperty(attributeName, SVGAnimatedNumberAccessor<OwnerType>::template singleton<property, initialValue>());
+        registerProperty(attributeName, SVGAnimatedNumberAccessor<OwnerType>::template singleton<property>(initialValue));
     }
 
     template<const LazyNeverDestroyed<const QualifiedName>& attributeName, const Ref<SVGAnimatedNumberList> OwnerType::*property>
@@ -167,16 +167,16 @@ public:
         registerProperty(attributeName, SVGAnimatedTransformListAccessor<OwnerType>::template singleton<property>());
     }
 
-    template<const LazyNeverDestroyed<const QualifiedName>& attributeName, const Ref<SVGAnimatedInteger> OwnerType::*property1, const Ref<SVGAnimatedInteger> OwnerType::*property2, int initialValue1 = 0, int initialValue2 = 0>
-    static void registerProperty()
+    template<const LazyNeverDestroyed<const QualifiedName>& attributeName, const Ref<SVGAnimatedInteger> OwnerType::*property1, const Ref<SVGAnimatedInteger> OwnerType::*property2>
+    static void registerProperty(int initialValue1 = 0, int initialValue2 = 0)
     {
-        registerProperty(attributeName, SVGAnimatedIntegerPairAccessor<OwnerType>::template singleton<property1, property2, initialValue1, initialValue2>());
+        registerProperty(attributeName, SVGAnimatedIntegerPairAccessor<OwnerType>::template singleton<property1, property2>(initialValue1, initialValue2));
     }
 
-    template<const LazyNeverDestroyed<const QualifiedName>& attributeName, const Ref<SVGAnimatedNumber> OwnerType::*property1, const Ref<SVGAnimatedNumber> OwnerType::*property2, float initialValue1 = 0.0f, float initialValue2 = 0.0f>
-    static void registerProperty()
+    template<const LazyNeverDestroyed<const QualifiedName>& attributeName, const Ref<SVGAnimatedNumber> OwnerType::*property1, const Ref<SVGAnimatedNumber> OwnerType::*property2>
+    static void registerProperty(float initialValue1 = 0, float initialValue2 = 0)
     {
-        registerProperty(attributeName, SVGAnimatedNumberPairAccessor<OwnerType>::template singleton<property1, property2, initialValue1, initialValue2>());
+        registerProperty(attributeName, SVGAnimatedNumberPairAccessor<OwnerType>::template singleton<property1, property2>(initialValue1, initialValue2));
     }
 
     // Enumerate all the SVGMemberAccessors recursively. The functor will be called and will
-- 
2.53.0

